dev/AppLifecycle/AppInstance.cpp

```diff
@@ -333,6 +338,95 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
+        }
+
+        // For packaged, redirect to the already existing API.
+        if (IsPackagedProcess())
+        {
+            return winrt::Windows::ApplicationModel::Core::CoreApplication::RequestRestartAsync(arguments).get();
+        }
+
+        // Only one restart can happen at a time.
+        auto mutexName = wil::str_printf<std::wstring>(L"%s_RequestRestartNowInProgress", ComputeAppId().c_str());
+        wil::unique_mutex restartMutex;
+
+        restartMutex.create(mutexName.c_str(), CREATE_MUTEX_INITIAL_OWNER, MUTEX_ALL_ACCESS);
+        if (GetLastError() == ERROR_ALREADY_EXISTS)
+        {
+            return AppRestartFailureReason::RestartPending;
+        }
+
+        auto releaseOnExit{ restartMutex.ReleaseMutex_scope_exit() };
+
+        // Use DuplicateHandle to get a real handle from the pseudo-handle returned by GetCurrentProcess().  A real handle
+        // is required in order for it to be inherited 
+        wil::unique_handle parentHandle;
+        THROW_IF_WIN32_BOOL_FALSE(DuplicateHandle(GetCurrentProcess(), GetCurrentProcess(), GetCurrentProcess(), wil::out_param(parentHandle), 
+            PROCESS_QUERY_INFORMATION | SYNCHRONIZE | PROCESS_TERMINATE, TRUE, 0));
+
+        auto exePath = GenerateRestartAgentPath();
+
+        // c:\currentdirectory\restartagent.exe <inherited handle id of calling process> <custom arguments passed by caller>
+        auto cmdLine = wil::str_printf<wil::unique_cotaskmem_string>(L"\"%s\" %d %s", exePath.c_str(), parentHandle.get(), arguments.c_str());
+
+        // Launch the restart agent and explicitly inherit the current process' handle to it.  This allows the restart agent to
+        // sniff out the exact path of the caller executable in a sane way.
+        SIZE_T attributeListSize{ 0 };
+        THROW_HR_IF(E_UNEXPECTED, InitializeProcThreadAttributeList(nullptr, 1, 0, &attributeListSize));
+        THROW_LAST_ERROR_IF(GetLastError() != ERROR_INSUFFICIENT_BUFFER);
+
+        wil::unique_process_heap_ptr<_PROC_THREAD_ATTRIBUTE_LIST> attributeList(reinterpret_cast<PPROC_THREAD_ATTRIBUTE_LIST> (HeapAlloc(GetProcessHeap(), 0, attributeListSize)));
+        THROW_IF_NULL_ALLOC(attributeList);
+
+        THROW_IF_WIN32_BOOL_FALSE(InitializeProcThreadAttributeList(attributeList.get(), 1, 0, &attributeListSize));
+        auto freeAttributeList = wil::scope_exit([&] { DeleteProcThreadAttributeList(attributeList.get()); });
+
+        size_t handlesToInheritCount{ 1 };
+        HANDLE* handlesToInherit = reinterpret_cast<HANDLE*>(parentHandle.addressof());
+        THROW_IF_WIN32_BOOL_FALSE(UpdateProcThreadAttribute(attributeList.get(), 0, PROC_THREAD_ATTRIBUTE_HANDLE_LIST, handlesToInherit, 
+            handlesToInheritCount * sizeof(HANDLE), nullptr, nullptr));
+
+        STARTUPINFOEX info{};
+        info.StartupInfo.cb = sizeof(info);
+        info.lpAttributeList = attributeList.get();
+        
+        PROCESS_INFORMATION processInfo{};
```

Comment:
wil::unique_process_info?

Reply:
?

Reply:
I think you mean unique_process_information.